docs/samples/Microsoft.ML.Samples/Microsoft.ML.Samples.csproj

```diff
@@ -12,13 +12,15 @@
     <ProjectReference Include="..\..\..\src\Microsoft.ML.FastTree\Microsoft.ML.FastTree.csproj" />
     <ProjectReference Include="..\..\..\src\Microsoft.ML.LightGBM\Microsoft.ML.LightGBM.csproj" />
     <ProjectReference Include="..\..\..\src\Microsoft.ML.Recommender\Microsoft.ML.Recommender.csproj" />
+    <ProjectReference Include="..\..\..\src\Microsoft.ML.TensorFlow\Microsoft.ML.TensorFlow.csproj" />
     <ProjectReference Include="..\..\..\src\Microsoft.ML.TimeSeries\Microsoft.ML.TimeSeries.csproj" />
     <ProjectReference Include="..\..\..\src\Microsoft.ML.OnnxTransform\Microsoft.ML.OnnxTransform.csproj" />
 
     <NativeAssemblyReference Include="CpuMathNative" />
     <NativeAssemblyReference Include="FastTreeNative" />
     <NativeAssemblyReference Include="MatrixFactorizationNative" />
     <NativeAssemblyReference Include="LdaNative" />
+    <PackageReference Include="Microsoft.ML.TensorFlow.Redist" Version="0.7.0" />
```

Comment:
><PackageReference Include="Microsoft.ML.TensorFlow.Redist" Version="0.7.0" /> [](start = 4, length = 77)

Is this really necessary?
I've look in our test projects, and we don't reference it.

Reply:
This was a bit surprising for me too. Without adding the Microsoft.ML.TensorFlow.Redist  nuget, the Samples project complains  saying it does not find tensorflow.dll

However, as you pointed out, the test project does not reference the Microsoft.ML.TensorFlow.Redist  nuget.


---
In reply to: [238764544](https://github.com/dotnet/machinelearning/pull/1812#discussion_r238764544) [](ancestors = 238764544)

Reply:
Discussed with Abhishek offline and figured out how to point to the tensorflow dll.


---
In reply to: [238786038](https://github.com/dotnet/machinelearning/pull/1812#discussion_r238786038) [](ancestors = 238786038,238764544)

Reply:
The test project has a `NativeAssemblyReference` which pulls in the dlls.  Instead of doing that, we are using the TensorFlow.Redist  package inside the Samples project.

@yaeldekel .  I tried using the TensorFlow nuget instead  of the Redist.  However that leads to an issue -  the  TensorFlow nuget  has dependency on the latest "official" ML.NET nuget  (0.8).  Code-changes  which have gone in after the  latest official release  start complaining during build.

For now, I am keeping this as-is i.e.  take nuget dependency on the  Redist package + add project reference to the Microsoft.ML.TensorFlow.csproj

 

---
In reply to: [239576832](https://github.com/dotnet/machinelearning/pull/1812#discussion_r239576832) [](ancestors = 239576832,238786038,238764544)